### DISCUSSION TITLE
how can I send a POST and/or PUT with multipart/mixed (not multipart/form-data)

### DISCUSSION THREAD

**Discussion by dtrucken** [axios/axios#5588] [Q&A]
Question: is there a way to send a multipart PUT with a value other than what gets hard-coded in lib/core/Axios.js line 184
'Content-Type': 'multipart/form-data'

In versions of axis 0.26.x and earlier, I could change that to what my 3rd-party server requires: 'multipart/mixed', but now I can not. I would like to use the 1.x.x version to get past all the security issues flagged in earlier versions.

**Comment by DigitalBrainJS** [comment]
> In versions of axis 0.26.x and earlier, I could change that to what my 3rd-party server requires: 'multipart/mixed', but now I can not.

Axios 0.26 and later ignores the content type provided by the user for the FormData payload, browsers set this header themselves. So I'm not sure how it could work for you.
Automatic serialization is only performed on simple object payloads with a "multipart/form-data" content type.

**Reply by dtrucken** [reply]
Thanks.  I'm using node, there is no browser involved.  I cannot use the automatic serialization; because I can't use multipart/form-data, my 3rd-party server requires multipart/mixed.  In < 0.26.x I was able to use FormData() and pass in a filestream and it did all the multi-part processing for me.  That appears not to work anymore.  Is there a way to create the multi-part form then use multipart/mixed and have the multi-part processing work?  Or do I need to manually code all the file parts?

To be clear, my code is using node (written in typescript) to upload files on the disk of my docker container to a 3rd-party server.  There are no humans or browsers involved.

**Reply by DigitalBrainJS** [reply]
Unfortunately, the old custom types are not supported. You can use hacks like converting the form data stream to a buffer (form.getBuffer()), proxying the stream with `stream.PassThrough`, or subclassing FormData and rewriting the toString method to avoid Form detection which will cause default headers to be applied.
I will consider adding support for `multipart/` header rewriting.

https://codesandbox.io/p/sandbox/axios-promise-fetch-node-forked-okb0yk

```
import axios from "axios";
import FormData from "form-data";

class MyFormData extends FormData {
  toString() {}
}

const form = new MyFormData();

form.append("x", 123);

const { data } = await axios.post("http://httpbin.org/post", form, {
  headers: {
    ...form.getHeaders(),
    "Content-Type": "multipart/mixed; boundary=" + form.getBoundary(),
  },
});

console.log(data);

```

**Reply by dtrucken** [reply]
Thanks for the reply, but nothing seems to work for me.  Specifically, FormData no longer supports a filestream.  Blob would slurp the entire file into memory.

Are you planning support for multipart/ header rewriting?  If so, where can I vote for it?
